com.minres.coredsl/src/com/minres/coredsl/scoping/CoreDslScopeProvider.xtend

```diff
@@ -0,0 +1,258 @@
+			}
+			UnionTypeSpecifier: {
+				val typeDeclaration = typeSpecifier.target;
+				if(typeDeclaration === null) return null
+
+				return typeDeclaration.members.flatMap[it.declarators];
+			}
+			default: {
+				return null;
+			}
+		}
+	}
+
+	def private static IScope getTypeMemberScope(TypeSpecifier typeSpecifier) {
+		val members = getTypeMembers(typeSpecifier);
+		return members !== null ? Scopes.scopeFor(members) : IScope.NULLSCOPE;
+	}
+
+	def private static IScope getMemberScope(EObject context) {
+		val memberAccess = context.castOrNull(MemberAccessExpression);
+		if(memberAccess === null) return IScope.NULLSCOPE;
+
+		val typeSpecifier = findTypeSpecifier(memberAccess.target);
+		if(typeSpecifier === null) return IScope.NULLSCOPE;
+
+		return getTypeMemberScope(typeSpecifier);
+	}
+
+	def private static IScope getParentNamedEntityScope(EObject context) {
+		return getNamedEntityScope(context.eContainer, context);
+	}
+
+	def private static dispatch IScope getNamedEntityScope(ForLoop context, EObject child) {
+		if(child !== null && child.eContainingFeature === CoreDslPackage.Literals.FOR_LOOP__START_DECLARATION) {
+			return getParentNamedEntityScope(context);
+		}
+
+		if(context.startDeclaration !== null) {
+			return Scopes.scopeFor(context.startDeclaration.declarators, getParentNamedEntityScope(context));
+		} else {
+			return Scopes.scopeFor(#[], getParentNamedEntityScope(context));
+		}
+	}
+
+	def private static dispatch IScope getNamedEntityScope(Declaration context, EObject child) {
+		return Scopes.scopeFor(context.declarators.takeWhile[it !== child], getParentNamedEntityScope(context));
+	}
+
+	def private static Iterable<NamedEntity> getDeclarationsTransitive(InstructionSet iset) {
+		return getDeclarationsTransitive(iset, new HashSet());
+	}
+
+	def private static Iterable<NamedEntity> getDeclarationsTransitive(InstructionSet isa,
+		HashSet<InstructionSet> seen) {
+		if(!seen.add(isa)) return #[]; // recursion error
+		
+		val declarations = isa.archStateBody.filter(DeclarationStatement).flatMap[it.declaration.declarators];
+		val enumMembers = isa.types.filter(EnumTypeDeclaration).flatMap[it.members];
+		val functions = isa.functions;
+		
+		val entities = declarations + enumMembers + functions;
+		return isa.superType !== null ? entities + getDeclarationsTransitive(isa.superType) : entities;
+	}
+
+	def private static dispatch IScope getInheritedScope(CoreDef core) {
+		val declarations = core.providedInstructionSets.flatMap[getDeclarationsTransitive(it)];
+		val seen = new HashSet();
+		return Scopes.scopeFor(declarations.filter[seen.add(it)]);
+	}
+
+	def private static dispatch IScope getInheritedScope(InstructionSet iset) {
+		if(iset.superType === null) return IScope.NULLSCOPE;
+		Scopes.scopeFor(getDeclarationsTransitive(iset.superType))
+	}
+
+	def private static dispatch IScope getNamedEntityScope(ISA context, EObject child) {
+		val declarations = context.archStateBody //
```

Comment:
Empty comments?

Reply:
They are there because the XTend formatter doesn't have an option for "keep manual line breaks", so it tries to join all of those lines.